### PR TITLE
Embed maker/taker frontend into binary

### LINKED ISSUE #64: Embed frontend in binaries
Automate embedding of frontend artifacts in the respective binaries and serve them on the same port as the HTTP API.

Outcome should be a self-contained binary that serves its own frontend once started.

### PR DESCRIPTION
**Pull request by bonomat**
Note: we need to check-in `frontend/dist` directories as otherwise the build will fail.
If you want to embed the files into the binaries you will need to build them first (`yarn run build`).

resolves #64

### CODE REVIEW

File: daemon/src/routes_maker.rs [hunk 1]
```diff
@@ -4,11 +4,15 @@ use crate::model::{Usd, WalletInfo};
 use crate::to_sse_event::ToSseEvent;
 use anyhow::Result;
```
Comment (thomaseizinger) [inline daemon/src/routes_maker.rs]: Can we remove this newline? rustfmt doesn't format import blocks across empty lines.

File: daemon/src/routes_maker.rs [hunk 2]
```diff
@@ -106,3 +110,32 @@ pub fn get_health_check() {}
...
+                .extension()
+                .and_then(OsStr::to_str)
+                .ok_or_else(|| Status::new(400))?;
+            let content_type = ContentType::from_extension(ext).ok_or_else(|| Status::new(400))?;
+            Ok((content_type, d.data))
+        },
+    )
+}
+
+#[rocket::get("/<_paths..>")]
```
Comment (thomaseizinger) [inline daemon/src/routes_maker.rs]: Can we add `format = "text/html"` here?

That would make it more obvious that we only want to respond with `index.html` for all requests that ask for a HTML page.

I think  that should be possible: https://docs.rs/rocket/0.5.0-rc.1/rocket/attr.get.html#grammar

File: daemon/src/routes_maker.rs [hunk 3]
```diff
@@ -106,3 +110,32 @@ pub fn get_health_check() {}
...
+pub fn index<'r>(_paths: PathBuf) -> impl Responder<'r, 'static> {
+    Asset::get("index.html").map_or_else(
+        || Err(Status::NotFound),
+        |d| Ok((ContentType::HTML, d.data)),
```
Comment (thomaseizinger) [inline daemon/src/routes_maker.rs]: I'd find this easier to read (untested but I think should work):

```suggestion
    let asset = Asset::get("index.html").ok_or(Status::NotFound)?;
    
    Ok((ContentType::HTML, asset.data))
```

Or if you wanted to use one of rocket's built-in responders:

```rust
Ok(Html(asset.data))
```

File: daemon/src/routes_maker.rs [hunk 4]
```diff
@@ -106,3 +110,32 @@ pub fn get_health_check() {}
...
+            let ext = file
+                .as_path()
+                .extension()
+                .and_then(OsStr::to_str)
+                .ok_or_else(|| Status::new(400))?;
+            let content_type = ContentType::from_extension(ext).ok_or_else(|| Status::new(400))?;
+            Ok((content_type, d.data))
+        },
+    )
+}
```
Comment (thomaseizinger) [inline daemon/src/routes_maker.rs]: These are exactly the same for both right?

Can we have a shared implementation that is generic over the `RustEmbed` trait and hence allows to plugin in the respective `Asset` struct?

**Review by thomaseizinger — APPROVED** [review]
Thanks!

Couple of nits :)

File: daemon/src/routes_maker.rs (hunk 4 above)
Comment (bonomat) [reply daemon/src/routes_maker.rs]: I gave it a try in : [`28cc157` (#95)](https://github.com/comit-network/hermes/pull/95/commits/28cc157cee7da73cbe9a17ca2bbe59b876d019dc)

**Review by thomaseizinger — APPROVED** [review]
Nice, thank you for making it more DRY!

**Pull request merged**